extensions/notification/NotificationQueuePopulator.js

```diff
@@ -19,110 +17,35 @@ class NotificationQueuePopulator extends QueuePopulatorExtension {
-                if (error) {
-                    this.log.error('invalid config', { method, zkPath, data });
-                    return done(null, 1);
-                }
-                this.log.debug('fetched bucket notification configuration', {
-                    method,
-                    zkPath,
-                    data: result,
-                });
-                return done(null, result);
-            }
-            return done(null, 0);
-        });
-    }
-
-    _setBucketNotifConfig(bucket, data, done) {
-        const method
-            = 'NotificationQueuePopulator._setBucketNotifConfig';
-        const zkPath = this._getBucketNodeZkPath(bucket);
-        return this.zkClient.setData(zkPath, Buffer.from(data), -1, err => {
-            if (err) {
-                this.log.error('error saving config', { method, zkPath, data });
-                return done(err);
-            }
-            this.log.debug('config saved', { method, zkPath, data });
-            return done();
-        });
-    }
-
-    _createBucketNotifConfigNode(bucket, done) {
-        const method
-            = 'NotificationQueuePopulator._createBucketNotifConfigNode';
-        const zkPath = this._getBucketNodeZkPath(bucket);
-        return this.zkClient.mkdirp(zkPath, err => {
-            if (err) {
-                this.log.error('Could not pre-create path in zookeeper', {
-                    method,
-                    zkPath,
-                    error: err,
-                });
-                return done(err);
-            }
-            return done();
-        });
-    }
-
-    _removeBucketNotifConfigNode(bucket, done) {
-        const method
-            = 'NotificationQueuePopulator._removeBucketNotifConfigNode';
-        const zkPath = this._getBucketNodeZkPath(bucket);
-        return this.zkClient.remove(zkPath, err => {
-            if (err && err.name !== 'NO_NODE') {
-                this.log.error('Could not remove zookeeper node', {
-                    method,
-                    zkPath,
-                    error: err,
-                });
-                return done(err);
-            }
-            if (!err) {
-                const errMsg
-                    = 'removed notification configuration zookeeper node';
-                this.log.info(errMsg, {
-                    method,
-                    bucket,
-                });
-            }
-            return done();
-        });
+        this.bnConfigManager = params.bnConfigManager;
+        assert(this.bnConfigManager, 'bucket notification configuration manager'
+            + ' is not set');
     }
 
+    /**
+     * Check if bucket entry based on bucket and key params
+     *
+     * @param {String} bucket - bucket
+     * @param {String} key - object key
+     * @return {boolean} - true if bucket entry
+     */
     _isBucketEntry(bucket, key) {
         return ((bucket.toLowerCase() === 'metastore' && key)
```

Comment:
Where does it come from `metastore` ? Sometimes it is `__metastore` (e.g. Mongo). Shall it be in `constants.js` ?

Reply:
The current implementation only targets S3C, notification is not implemented in zenko, so that's why only `metastore` is referenced. If your question is where `metastore` comes from, it is from the log, a sample entry,
`{ type: 'put',
  bucket: 'metastore',
  key: 'db/bucket1',
  value:
   '{"name":"bucket1","attributes":"{\\"name\\":\\"bucket1\\",\\"owner\\":\\"9abe0ec063308bc0c7cd5f99e1260f942afb7e8682a86edc807795fa652ee28a\\",\\"ownerDisplayName\\":\\"naren\\",\\"creationDate\\":\\"2020-08-30T02:42:48.224Z\\",\\"mdBucketModelVersion\\":8,\\"transient\\":true,\\"deleted\\":false,\\"serverSideEncryption\\":null,\\"versioningConfiguration\\":null,\\"locationConstraint\\":\\"us-east-1\\",\\"cors\\":null,\\"replicationConfiguration\\":null,\\"lifecycleConfiguration\\":null,\\"bucketPolicy\\":null,\\"objectLockEnabled\\":false,\\"objectLockConfiguration\\":null,\\"notificationConfiguration\\":null}","creating":true,"deleting":false,"id":"bucket1","raftSessionID":1,"version":2}' }`

Reply:
Ref, https://github.com/scality/Federation/commit/351b022151b52a73f5a6ac176031f5f767c69c44